doc/source/data/faq.rst

```diff
@@ -0,0 +1,318 @@
+* TensorFlow distributed datasets provide
+  `minimal data sharding control <https://www.tensorflow.org/tutorials/distribute/input#sharding>`__,
+  only providing file-based sharding and naive pull-all-and-drop sharding; meanwhile,
+  Ray Datasets allows you to perform arbitrary sharding logic.
+* TensorFlow datasets parallel data loading and last-mile preprocessing is limited to
+  single-node multi-processing, which has many issues that
+  `Ray solves
+  <https://towardsdatascience.com/modern-parallel-and-distributed-python-a-quick-tutorial-on-ray-99f8d70369b8>`__
+  , in particular zero-copy exchange between the same-node transforming/prefetching
+  process and the trainer process is particularly valuable. Ray Datasets exposes these
+  core Ray advantages in convenient parallelization APIs and adds additional
+  optimizations around data batching and memory management.
+* Anything beyond basic/cheap preprocessing requires stitching together another
+  framework like Spark or Dask with this TensorFlow dataset, which adds infrastructure
+  complexity and can break pipelining of loading + preprocessing + training. With Ray
+  Datasets, you can express complex last-mile preprocessing operations that are optimally
+  pipelined with training.
+* When doing data-parallel training, TensorFlow datasets do not support full-shuffling
+  across worker shards on every epoch (only file interleaving is supported, not full
+  data mixing), while Ray Datasets supports this both in the single-node case and the
+  distributed data-parallel case.
+
+Petastorm
+~~~~~~~~~
+
+* `Petastorm <https://github.com/uber/petastorm>`__ only supports Parquet data, while
+  Ray Datasets supports many file formats.
+* Petastorm uses a multi-processing data loader, which has many issues that
+  `Ray solves
+  <https://towardsdatascience.com/modern-parallel-and-distributed-python-a-quick-tutorial-on-ray-99f8d70369b8>`__
+  , in particular zero-copy exchange between the same-node prefetching
+  process and the trainer process is particularly valuable. Ray Datasets exposes these
+  core Ray advantages in convenient parallelization APIs and adds additional
+  optimizations around data batching and memory management.
+* Petastorm does not expose any data processing APIs.
+
+NVTabular
+~~~~~~~~~
+
+* `NVTabular <https://github.com/NVIDIA-Merlin/NVTabular>`__ only supports tabular
+  (Parquet, CSV, Avro) data, while Ray Datasets supports many other file formats.
+* NVTabular doesn't support mixing heterogeneous resources in dataset transforms (e.g.
+  both CPU and GPU transformations), while Ray Datasets supports this.
+* NVTabular has a bunch of great ML-specific preprocessing operations; this is current
+  WIP for Ray Datasets via the :ref:`Ray AIR preprocessors <air-key-concepts>`.
+
+For batch (offline) inference, why should I use Ray Datasets instead of an actor pool?
+======================================================================================
```

Comment:
That question is indeed more important, and this one is more like if user is already in Ray ecosystem, why they should use Dataset rather than DIY with Ray's lower-level APIs. For the first version, it sounds fine to drop this question.